goyacc/format_yacc.go

```diff
@@ -0,0 +1,581 @@
+		case LCURLRCURLCase:
+			err = handleProlog(formatter, def)
+		case ReservedWordTagNameListCase:
+			err = handleReservedWordTagNameList(formatter, def)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	_, err := formatter.Format("\n%%%%")
+	return err
+}
+
+func handleStart(f format.Formatter, definition *parser.Definition) error {
+	if err := Ensure(definition).
+		and(definition.Token2).
+		and(definition.Token2).NotNil(); err != nil {
+		return err
+	}
+	cmt1 := strings.Join(definition.Token.Comments, "\n")
+	cmt2 := strings.Join(definition.Token2.Comments, "\n")
+	_, err := f.Format("\n%s%s\t%s%s\n", cmt1, definition.Token.Val, cmt2, definition.Token2.Val)
+	return err
+}
+
+func handleUnion(f format.Formatter, definition *parser.Definition) error {
+	if err := Ensure(definition).
+		and(definition.Value).NotNil(); err != nil {
+		return err
+	}
+	if len(definition.Value) != 0 {
+		_, err := f.Format("%%union%i%s%u\n\n", definition.Value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func handleProlog(f format.Formatter, definition *parser.Definition) error {
+	if err := Ensure(definition).
+		and(definition.Value).NotNil(); err != nil {
+		return err
+	}
+	_, err := f.Format("%%{%s%%}\n\n", definition.Value)
+	return err
+}
+
+func handleReservedWordTagNameList(f format.Formatter, def *parser.Definition) error {
+	if err := Ensure(def).
+		and(def.ReservedWord).
+		and(def.ReservedWord.Token).NotNil(); err != nil {
+		return err
+	}
+	comment := getTokenComment(def.ReservedWord.Token, divNewLineStringLayout)
+	directive := def.ReservedWord.Token.Val
+
+	hasTag := def.Tag != nil
+	var wordAfterDirective string
+	if hasTag {
+		wordAfterDirective = joinTag(def.Tag)
+	} else {
+		wordAfterDirective = joinNames(def.Nlist)
+	}
+
+	if _, err := f.Format("%s%s%s%i", comment, directive, wordAfterDirective); err != nil {
+		return err
+	}
+	if hasTag {
+		if _, err := f.Format("\n"); err != nil {
+			return err
+		}
+		if err := printNameListVertical(f, def.Nlist); err != nil {
+			return err
+		}
+	}
+	_, err := f.Format("%u\n")
+	return err
+}
+
+func joinTag(tag *parser.Tag) string {
+	var sb strings.Builder
+	sb.WriteString("\t")
+	if tag.Token != nil {
+		sb.WriteString(tag.Token.Val)
+	}
+	if tag.Token2 != nil {
+		sb.WriteString(tag.Token2.Val)
+	}
+	if tag.Token3 != nil {
+		sb.WriteString(tag.Token3.Val)
+	}
+	return sb.String()
+}
+
+type stringLayout int8
+
+const (
+	spanStringLayout stringLayout = iota
+	divStringLayout
+	divNewLineStringLayout
+)
+
+func getTokenComment(token *parser.Token, layout stringLayout) string {
+	if len(token.Comments) == 0 {
+		return ""
+	}
+	var splitter, beforeComment string
+	switch layout {
+	case spanStringLayout:
+		splitter, beforeComment = " ", ""
+	case divStringLayout:
+		splitter, beforeComment = "\n", ""
+	case divNewLineStringLayout:
+		splitter, beforeComment = "\n", "\n"
+	default:
+		panic(errors.WithStack(errors.Errorf("unsupported stringLayout: %v", layout)))
```

Comment:
```suggestion
		panic(errors.Errorf("unsupported stringLayout: %v", layout))
```

Errorf should contain the stack already.